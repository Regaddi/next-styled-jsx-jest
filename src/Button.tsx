import * as React from "react";

interface ButtonProps {
  size: "small" | "medium";
}

const sizes = {
  small: "12px",
  medium: "16px"
};

export const Button: React.FC<ButtonProps> = ({ children, size, ...props }) => {
  return (
    <>
      <button type="button" {...props}>
        {children}
      </button>
      <style jsx>
        {`
          button {
            font-size: ${sizes[size]};
          }
        `}
      </style>
    </>
  );
};
