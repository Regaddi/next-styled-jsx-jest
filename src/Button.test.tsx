import * as React from "react";
import { render } from "@testing-library/react";
import { Button } from "./Button";

import "@testing-library/jest-dom/extend-expect";

describe("Button", () => {
  it("renders correctly", () => {
    const { container } = render(<Button size="small">Click me</Button>);
    expect(container).toMatchSnapshot();
  });
});
