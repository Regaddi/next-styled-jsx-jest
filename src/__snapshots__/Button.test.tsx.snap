// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<div>
  <button
    type="button"
  >
    Click me
  </button>
  <style>
    
          button {
            font-size: 12px;
          }
        
  </style>
</div>
`;
